### ISSUE TITLE
Single/double quote code style

### ISSUE THREAD

**Issue by yakky** [divio/django-cms#4961]
Currently python codebase is a random mix of single and double quotes. Do we want to set a policy for this?
Frontend already has its style guideline.
I'd be in favor of single quotes everywhere except multiline strings

**Comment by yakky** [comment]
Any opinion on this ? @timgraham @evildmp @mkoistinen @czpython @FinalAngel @vxsx

**Comment by timgraham** [comment]
I usually go with single quotes except if the string contains a single quote -- then use double quotes to avoid having to escape the quote. Agreed with double quotes for multiline strings and docstrings.

**Comment by FinalAngel** [comment]
We use double quotes for markup / css and single quotes for coding / javascript: https://aldryn-boilerplate-bootstrap3.readthedocs.org/en/latest/guidelines/general.html#quotes

This includes Django template tags, for example ``{% trans "My Text" %}`` and not ``{% trans 'My Text' %}``. I would prefer single quotes in Django / Python as well, though a lot of tutorials and example mix them. When using single quotes, double quotes are allowed inside like: ``print 'This my my "example"';``

TLDR: single quotes

**Comment by yakky** [comment]
@finalangel I edited the description to be clear that I was referring to python code. Frontend already has its policy in place

**Comment by mkoistinen** [comment]
I prefer single quotes, except where you need to otherwise escape an apostrophe, etc. So, +1 for me. On my keyboard layout, a single-quote just requires fewer keystrokes and they're a little easier on the eyes.

**Comment by FinalAngel** [comment]
@yakky I refer to python code: single quotes. Just making sure that for template tags, it's clear to use double quotes.

**Comment by mkoistinen** [comment]
@evildmp I think we have enough approvals here to ratify this in a core devs meeting? I'm closing this here, but let's record this as "accepted" somewhere.

**Comment by tony** [comment]
I know this is old, but just for future reference: (i've been trying to see what larger projects are using)

There is a neat plugin to enforce quotes with flake8, https://github.com/zheller/flake8-quotes

**Comment by jrief** [comment]
Just to add my two cents. I like the proposal of Douglas Crockford (the author of "JavaScript: The Good Parts") for this:

* Use single quotes to specify strings which are used by the program internally.
* Use double quotes to expose strings which are visible by humans.

This means that the following example is written as:

```
color = fields.ChoiceField(
    label="Color",
    choices=[('ff0000', "Red"), ('00ff00', "Green"), ('0000ff', "Blue")]),
)
```

It allows us to determine all strings which might have to be translated. Since apostrophes usually are only used inside text read by humans, this is handled as well.
